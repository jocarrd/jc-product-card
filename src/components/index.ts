import { ProductCardHOCProps } from "../interfaces/interfaces";
import { ProductButtons } from "./ProductButtons";
import { PoductCard as ProductCardHOC } from "./ProductCard";
import { ProductImage } from "./ProductImage";
import { ProductTitle } from "./ProductTitle";

export{ ProductButtons } from "./ProductButtons";
export { ProductTitle } from "./ProductTitle";
export { ProductImage } from "./ProductImage";


export const ProductCard : ProductCardHOCProps = Object.assign(ProductCardHOC,{
    Title:ProductTitle,
    Image : ProductImage,
    Buttons: ProductButtons
})

