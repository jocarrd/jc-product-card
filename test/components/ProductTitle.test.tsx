import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTitle', () => {
  test('Product Tile must be render with custom title', () => {
    const component = renderer.create(<ProductTitle title="Cutstom Product" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('ProductTitle must show', () => {
    const component = renderer.create(
      <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
