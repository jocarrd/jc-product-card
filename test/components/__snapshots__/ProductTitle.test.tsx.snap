// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Product Tile must be render with custom title 1`] = `
<span
  className="productDescription undefined"
>
  Cutstom Product
</span>
`;

exports[`ProductTitle ProductTitle must show 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Cofee-Mug without Image
  </span>
</div>
`;
